openfe/setup/methods/openmm/equil_rbfe_methods.py

```diff
@@ -0,0 +1,823 @@
+            'settings': dict(self._settings),
+        }
+
+    @classmethod
+    def from_dict(cls, d: dict):
+        """Deserialize from a dict representation"""
+        return cls(
+            stateA=ChemicalState.from_dict(d['stateA']),
+            stateB=ChemicalState.from_dict(d['stateB']),
+            mapping=LigandAtomMapping.from_dict(d['mapping']),
+            settings=dict(**d['settings']),
+        )
+
+    def run(self, dry=False, verbose=True) -> bool:
+        """Run the relative free energy calculation.
+
+        Parameters
+        ----------
+        dry : bool
+          Do a dry run of the calculation, creating all necessary hybrid
+          system components (topology, system, sampler, etc...) but without
+          running the simulation.
+
+        verbose : bool
+          Verbose output of the simulation progress.
+
+        Returns
+        -------
+        bool
+          True if everything went well.
+        """
+        stateA_openff_ligand = self._stateA.components['ligand'].to_openff()
+        stateB_openff_ligand = self._stateB.components['ligand'].to_openff()
+
+        ## Get smirnoff template generators
+        smirnoff_stateA = SMIRNOFFTemplateGenerator(
+            forcefield=self._settings.topology_settings.forcefield['ligand'],
+            molecules=[stateA_openff_ligand],
+        )
+
+        smirnoff_stateB = SMIRNOFFTemplateGenerator(
+            forcefield=self._settings.topology_settings.forcefield['ligand'],
+            molecules=[stateB_openff_ligand],
+        )
+
+        ## Create forece fields and register them
+        # state A
+        omm_forcefield_stateA = app.ForceField(
+            *[ff for (comp, ff) in self._settings.topology_settings.forcefield.items()
+              if not comp == 'ligand']
+        )
+
+        omm_forcefield_stateA.registerTemplateGenerator(
+                smirnoff_stateA.generator)
+
+        # state B
+        omm_forcefield_stateB = app.ForceField(
+            *[ff for (comp, ff) in self._settings.topology_settings.forcefield.items()
+              if not comp == 'ligand']
+        )
+
+        omm_forcefield_stateB.registerTemplateGenerator(
+                smirnoff_stateB.generator)
+
+
+        ##  Model state A
+        # todo - add logic for PDBFile once it's been fixed upstream
+        # if 'protein' in self._stateA.components:
+        #     pdbfile = self._stateA.components['protein'].to_openmm_PDBFile()
+        #     stateA_modeller = app.Modeller(pdbfile.topology,
+        #                                    pdbfile.positions)
+        #     stateA_modeller.add(
+        #         stateA_openff_ligand.to_topology().to_openmm(),
+        #         stateA_openff_ligand.conformers[0],
+        #     )
+        # else:
+        # --- add below --
+        stateA_modeller = app.Modeller(
+            stateA_openff_ligand.to_topology().to_openmm(),
+            stateA_openff_ligand.conformers[0],
+        )
```

Comment:
Once https://github.com/OpenFreeEnergy/gufe/pull/17 is merged.

```suggestion
        if 'protein' in self._stateA.components:
            pdbfile = self._stateA.components['protein'].to_openmm_PDBFile()
            stateA_modeller = app.Modeller(pdbfile.topology,
                                           pdbfile.positions)
            stateA_modeller.add(
                stateA_openff_ligand.to_topology().to_openmm(),
                stateA_openff_ligand.conformers[0],
            )
        else:
            stateA_modeller = app.Modeller(
                stateA_openff_ligand.to_topology().to_openmm(),
                stateA_openff_ligand.conformers[0],
            )
```